### PR TITLE
light: wait for tendermint node to start before running example test

### PR DESCRIPTION
**Pull request by cmwaters**
Closes: #5999

### CODE REVIEW

File: light/light_test.go [hunk 1]
```diff
@@ -136,3 +142,18 @@ func TestClientIntegration_VerifyLightBlockAtHeight(t *testing.T) {
...
+
+func waitForBlock(ctx context.Context, p provider.Provider, height int64) (*types.LightBlock, error) {
+	for {
+		block, err := p.LightBlock(ctx, height)
+		switch err {
+		case nil:
+			return block, nil
+		// node isn't running yet, wait 1 second and repeat
+		case provider.ErrNoResponse, provider.ErrHeightTooHigh:
+			time.Sleep(1 * time.Second)
```
Comment (tychoish) [inline light/light_test.go:154]: I'd do this as a ticker/timer and have this happen inside of a select statement with the timer/ticker and the ctx.Done() in different cases.

File: light/light_test.go [hunk 2]
```diff
@@ -136,3 +142,18 @@ func TestClientIntegration_VerifyLightBlockAtHeight(t *testing.T) {
 
 	require.EqualValues(t, 3, h.Height)
 }
+
+func waitForBlock(ctx context.Context, p provider.Provider, height int64) (*types.LightBlock, error) {
```
Comment (tychoish) [inline light/light_test.go:146]: why not make this a public function or a method on provider? it it's used in the example then people will need to go hunting for it.

File: light/example_test.go [hunk 3]
```diff
@@ -81,91 +79,33 @@ func ExampleClient_Update() {
...
+	h, err = c.Update(ctx, time.Now())
 	if err != nil {
 		stdlog.Fatal(err)
 	}
 
-	h, err := c.TrustedLightBlock(3)
-	if err != nil {
-		stdlog.Fatal(err)
+	if h != nil && h.Height > 3 {
+		fmt.Println("successful update")
```
Comment (williambanfield) [inline light/example_test.go:107]: Is there a reason these a printlns and not log statements?

Comment (cmwaters) [reply light/example_test.go:107]: wasn't my decision. I can switch to using the logger

File: light/light_test.go (hunk 2 above)
Comment (cmwaters) [reply light/light_test.go:146]: True but it's used as part of the testing scaffold to ensure that the Tendermint node has reached a sufficient height for the test to deterministically pass. As this isn't really a test and we're not really asserting anything we could remove it and just plainly run the light client functions

File: light/example_test.go (hunk 3 above)
Comment (alexanderbez) [reply light/example_test.go:107]: Let's use a logger if we can.

**Review by tychoish — APPROVED** [review]
this looks good. The Example tests (in `example_test.go` no longer run as part of the tests) and are only there for documentation purposes now (as long as they compile), which means the second example might be fine to leave as is.

**Comment by cmwaters** [comment]
The only difference between the two examples is that one client used the `Update()` method whilst the other used `VerifyLightBlockAtHeight()` so I just merged them

**Pull request merged**